backend/compact-connect/lambdas/license-data/tests/unit/test_event_batch_writer.py

```diff
@@ -0,0 +1,144 @@
+
+        def mock_put_items(Entries: List[dict]):  # pylint: disable=invalid-name
+            put_count.extend(Entries)
+            return {}
+
+        mock_client = MagicMock()
+        mock_client.put_events.side_effect = mock_put_items
+
+        with open('tests/resources/ingest/message.json', 'r') as f:
+            event = json.load(f)
+
+        with EventBatchWriter(client=mock_client) as writer:
+            # Send a bunch of messages, make sure each is sent
+            for _ in range(123):
+                writer.put_event(
+                    Entry=event
+                )
+
+        # Make sure each message was eventually sent
+        self.assertEqual(123, len(put_count))
+        # Make sure these were sent in three batches
+        self.assertEqual(3, mock_client.put_events.call_count)
+
+    def test_write_small_batch(self):
+        from event_batch_writer import EventBatchWriter
+
+        put_count = []
+
+        def mock_put_items(Entries: List[dict]):  # pylint: disable=invalid-name
+            put_count.extend(Entries)
+            return {}
+
+        mock_client = MagicMock()
+        mock_client.put_events.side_effect = mock_put_items
+
+        with open('tests/resources/ingest/message.json', 'r') as f:
+            event = json.load(f)
+
+        with EventBatchWriter(client=mock_client) as writer:
+            # Send a bunch of messages, make sure each is sent
+            for _ in range(23):
+                writer.put_event(
+                    Entry=event
+                )
+
+        # Make sure each message was eventually sent
+        self.assertEqual(23, len(put_count))
+        # Make sure these were sent in one batch
+        self.assertEqual(1, mock_client.put_events.call_count)
+
+    def test_exception_recovery(self):
+        from event_batch_writer import EventBatchWriter
+
+        put_count = []
+
+        def mock_put_items(Entries: List[dict]):  # pylint: disable=invalid-name
+            put_count.extend(Entries)
+            return {}
+
+        mock_client = MagicMock()
+        mock_client.put_events.side_effect = mock_put_items
+
+        with open('tests/resources/ingest/message.json', 'r') as f:
+            event = json.load(f)
+
+        def interrupted_with_exception():
+            with EventBatchWriter(client=mock_client) as writer:
+                # Send a bunch of messages, make sure each is sent
+                for _ in range(23):
+                    writer.put_event(
+                        Entry=event
+                    )
+                raise RuntimeError('Oh noes!')
+
+        # Make sure the exception is not suppressed
+        with self.assertRaises(RuntimeError):
+            interrupted_with_exception()
+
+        # Make sure each message was eventually sent
+        self.assertEqual(23, len(put_count))
+        # Make sure these were sent in one batch
+        self.assertEqual(1, mock_client.put_events.call_count)
+
+    def test_bad_use(self):
```

Comment:
Hm, I this test is very terse but I guess that's actually taking away from clarity in this case. The 'bad use' here is not because of an invalid entry but because the `EventBatchWriter` itself is not being used as a context manager:

### Correct
```python
with EventBatchWriter(client=mock_client) as writer:
        writer.put_event(Entry=event)
```

### Incorrect
```python
writer = EventBatchWriter(client=mock_client)
writer.put_event(Entry=event)
```

So the latter should raise an exception, the former should not. I'll expand on the comments and method name so that is more clear.

Reply:
AH! That makes sense. Yeah, a little more of an explanatory comment would be great. (Even something similar to your comment here.) Thanks for the explanation.